lib/runtime_options.go

```diff
@@ -20,13 +20,22 @@
 
 package lib
 
-import null "gopkg.in/guregu/null.v3"
+import (
+	"fmt"
+	"strings"
+
+	"github.com/loadimpact/k6/js/compiler"
+	null "gopkg.in/guregu/null.v3"
+)
 
 // RuntimeOptions are settings passed onto the goja JS runtime
 type RuntimeOptions struct {
 	// Whether to pass the actual system environment variables to the JS runtime
 	IncludeSystemEnvVars null.Bool `json:"includeSystemEnvVars" envconfig:"K6_INCLUDE_SYSTEM_ENV_VARS"`
 
+	// JS compatibility mode: "extended" (Goja+Babel+core.js) or "base" (plain Goja)
+	CompatibilityMode null.String `json:"compatibilityMode"`
```

Comment:
As a nit to fix in the future (i.e. https://github.com/loadimpact/k6/issues/883), I'm wondering if the actual `CompatibilityMode` shouldn't be defined somewhere low-level (i.e. `lib` or wherever we decide to put these foundational types) and then instead of using a `string` here, we could just use that type (or some null-able wrapper of it).

Because right now, the stringy type here isn't very convenient. Even though it has been validated before we ever use it, we kind of have to cast it to the actual type if we ever want to use it, and then deal with the error that's never going to come...